modules/rabbitmq/src/main/java/org/testcontainers/containers/RabbitMQContainer.java

```diff
@@ -0,0 +1,334 @@
+                "tags=" + join(",", tags)));
+        return self();
+    }
+
+    public RabbitMQContainer withPolicy(String name, String pattern, Map<String, Object> definition) {
+        values.add(asList("rabbitmqadmin", "declare", "policy", "name=" + name, "pattern=" + pattern,
+                "definition=" + toJson(definition)));
+        return self();
+    }
+
+    public RabbitMQContainer withPolicy(String name, String pattern, Map<String, Object> definition, int priority, String applyTo) {
+        values.add(asList("rabbitmqadmin", "declare", "policy", "name=" + name,
+                "pattern=" + pattern, "priority=" + priority, "apply-to=" + applyTo, "definition=" + toJson(definition)));
+        return self();
+    }
+
+    public RabbitMQContainer withOperatorPolicy(String name, String pattern, Map<String, Object> definition) {
+        values.add(new ArrayList<>(asList("rabbitmqadmin", "declare", "operator_policy", "name=" + name,
+                "pattern=" + pattern, "definition=" + toJson(definition))));
+        return self();
+    }
+
+    public RabbitMQContainer withOperatorPolicy(String name, String pattern, Map<String, Object> definition, int priority, String applyTo) {
+        values.add(asList("rabbitmqadmin", "declare", "operator_policy", "name=" + name, "pattern=" + pattern,
+                "priority=" + priority, "apply-to=" + applyTo, "definition=" + toJson(definition)));
+        return self();
+    }
+
+    public RabbitMQContainer withVhost(String name) {
+        values.add(asList("rabbitmqadmin", "declare", "vhost", "name=" + name));
+        return self();
+    }
+
+    public RabbitMQContainer withVhost(String name, boolean tracing) {
+        values.add(asList("rabbitmqadmin", "declare", "vhost", "name=" + name, "tracing=" + tracing));
+        return self();
+    }
+
+    public RabbitMQContainer withVhostLimit(String vhost, String name, int value) {
+        values.add(asList("rabbitmqadmin", "declare", "vhost_limit", "vhost=" + vhost, "name=" + name, "value=" + value));
+        return self();
+    }
+
+    public RabbitMQContainer withQueue(String name) {
+        values.add(asList("rabbitmqadmin", "declare", "queue", "name=" + name));
+        return self();
+    }
+
+    public RabbitMQContainer withQueue(String name, boolean autoDelete, boolean durable, Map<String, Object> arguments) {
+        values.add(asList("rabbitmqadmin", "declare", "queue", "name=" + name,
+                "auto_delete=" + autoDelete, "durable=" + durable, "arguments=" + toJson(arguments)));
+
+        return self();
+    }
+
+    @NotNull
+    private String toJson(Map<String, Object> arguments) {
+        final ObjectMapper mapper = new ObjectMapper();
+        ObjectNode jsonObject = mapper.createObjectNode();
+        arguments.forEach((s, o) -> jsonObject.set(s, mapper.convertValue(o, JsonNode.class)));
+        return jsonObject.toString();
```

Comment:
why not just `return new ObjectMapper().writeValueAsString(arguments)`?

Reply:
Didn't know that method existed. Quite nice with a one-liner, but it does throw JsonProcessingException which needs to be handled. Nevertheless, I'll change it